File: GXpert/GXpert.Web/Modules/QuestionBank/QuestionCourse/QuestionCourseForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.QuestionBank.Forms;

[FormScript("QuestionBank.QuestionCourse")]
[BasedOnRow(typeof(QuestionCourseRow), CheckNames = true)]
public class QuestionCourseForm
{
    public int CourseId { get; set; }
    public int ClassId { get; set; }
    public int SemesterId { get; set; }
    public int SubjectId { get;set; }
    public int TopicId { get; set; }
    public long QuestionId { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/QuestionBank/QuestionCourseForm.ts
import { LookupEditor, ServiceLookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface QuestionCourseForm {
    CourseId: LookupEditor;
    ClassId: LookupEditor;
    SemesterId: ServiceLookupEditor;
    SubjectId: LookupEditor;
    TopicId: LookupEditor;
    QuestionId: LookupEditor;
}

export class QuestionCourseForm extends PrefixedContext {
    static readonly formKey = 'QuestionBank.QuestionCourse';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!QuestionCourseForm.init)  {
            QuestionCourseForm.init = true;

            var w0 = LookupEditor;
            var w1 = ServiceLookupEditor;

            initFormType(QuestionCourseForm, [
                'CourseId', w0,
                'ClassId', w0,
                'SemesterId', w1,
                'SubjectId', w0,
                'TopicId', w0,
                'QuestionId', w0
            ]);
        }
    }
}